### PR TITLE
Provide accessors to sigmask of UContext.

### PR DESCRIPTION
**Pull request by fiveop**

### CODE REVIEW

File: src/ucontext.rs [hunk 1]
```diff
@@ -22,4 +23,13 @@ impl UContext {
         };
         Errno::result(res).map(drop)
     }
+
+    #[allow(mutable_transmutes)]
+    pub fn mut_sigmask(&mut self) -> &mut SigSet {
```
Comment (fiveop) [inline src/ucontext.rs]: `mut_sigmask` or `sigmask_mut`?

Comment (fiveop) [inline src/ucontext.rs]: For `DerefMut` it's `deref_mut`, so I went with `sigmask_mut`.

Comment (fiveop) [inline src/ucontext.rs]: Another example that supports my choice of name:

https://github.com/rust-lang/rust/commit/9370d3a05107bdcac6fb62d3bd5380e1d7c79b5b

File: src/ucontext.rs [hunk 2]
```diff
@@ -22,4 +23,13 @@ impl UContext {
         };
         Errno::result(res).map(drop)
     }
+
+    #[allow(mutable_transmutes)]
```
Comment (kamalmarhubi) [inline src/ucontext.rs]: This attribute can be dropped if you do

```
unsafe { mem::transmute(&mut self.context.uc_sigmask) }
```

Comment (fiveop) [inline src/ucontext.rs]: Thanks you. I have missed that in some other code of mine as well. Any other comments?

**Comment by fiveop** [comment]
Can we merge this now?

**Comment by posborne** [comment]
Looks good.  @homu r+

**Pull request merged**